### PR TITLE
Feature - input autocomplete

### PR DESCRIPTION
**Pull request by JoaoDanielRufino**
### Description
Added input autocomplete(https://github.com/JoaoDanielRufino/go-input-autocomplete) to all workspaces input. Also, I added the option **type path** on config.json to include this input

### How to verify it
Run any formula with **type path** on config.json or any command that requires the user to enter the path to their workspaces. Hit **tab key** to autocomplete the path.
Example of config.json:
```
"inputs": [
    {
      "label": "Type your path: ",
      "name": "rit_path",
      "type": "path",
      "required": true
    }
  ]
```

### Changelog
Input Autocomplete

### CODE REVIEW

**Review by kaduartur — APPROVED** [review]
LGTM 🚀

File: pkg/cmd/add_workspace_test.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func TestNewAddWorkspaceCmd(t *testing.T) {
 
 			textMock := &mocks.InputTextMock{}
 			textMock.On("Text", "Enter the name of workspace", true, mock.Anything).Return(tt.workspaceName, nil)
-			textMock.On("Text", "Enter the path of workspace (e.g.: /home/user/github)", true, mock.Anything).Return(tt.workspacePath, nil)
+			// textMock.On("Text", "Enter the path of workspace (e.g.: /home/user/github) ", true, mock.Anything).Return(tt.workspacePath, nil)
```
Comment (henriquemoraeszup) [inline pkg/cmd/add_workspace_test.go:62]: commented code

File: pkg/formula/input/prompt/prompt_test.go [hunk 2]
```diff
@@ -939,3 +967,9 @@ func (fi fileManagerMock) Remove(path string) error {
 func (fi fileManagerMock) ListNews(oldPath, newPath string) ([]string, error) {
 	return fi.listNews, fi.lErr
 }
+
+type InputPathMock struct{}
+
+func (InputPathMock) Read(text string) (string, error) {
+	return text, nil
+}
```
Comment (henriquemoraeszup) [inline pkg/formula/input/prompt/prompt_test.go:975]: We should use testify mocks

File: pkg/formula/runner/docker/runner_test.go [hunk 3]
```diff
@@ -283,3 +283,9 @@ type inputResolverMock struct {
 func (i inputResolverMock) Resolve(inType api.TermInputType) (formula.InputRunner, error) {
 	return i.inRunner, i.err
 }
+
+type InputPathMock struct{}
+
+func (InputPathMock) Read(text string) (string, error) {
+	return text, nil
+}
```
Comment (henriquemoraeszup) [inline pkg/formula/runner/docker/runner_test.go:291]: we should strive to use testify mocks

File: pkg/formula/runner/local/runner_test.go [hunk 4]
```diff
@@ -271,3 +271,9 @@ type inputResolverMock struct {
 func (i inputResolverMock) Resolve(inType api.TermInputType) (formula.InputRunner, error) {
 	return i.inRunner, i.err
 }
+
+type InputPathMock struct{}
+
+func (InputPathMock) Read(text string) (string, error) {
+	return text, nil
+}
```
Comment (henriquemoraeszup) [inline pkg/formula/runner/local/runner_test.go:279]: same here

**Review by henriquemoraeszup — CHANGES_REQUESTED** [review]
Nice! just a couple of comments

File: pkg/cmd/mocks_test.go [hunk 5]
```diff
@@ -178,6 +178,12 @@ func (a repoListerAdderCustomMock) Add(d formula.Repo) error {
 	return a.add(d)
 }
 
+type InputPathMock struct{}
+
+func (InputPathMock) Read(text string) (string, error) {
+	return text, nil
+}
+
```
Comment (brunasilvazup) [inline pkg/cmd/mocks_test.go:186]: You can only use testify mocks in package mocks. We are changing all simulations for this.

File: pkg/formula/input/prompt/prompt_test.go [hunk 6]
```diff
@@ -752,6 +777,7 @@ func TestInputManager_Multiselect(t *testing.T) {
 			iBool := inputMock{boolean: false}
 			iPass := inputMock{text: "******"}
 			iMultiselect := inputMock{items: []string{}}
+			inPath := InputPathMock{}
```
Comment (brunasilvazup) [inline pkg/formula/input/prompt/prompt_test.go:780]: I was expecting specific tests for your new entry - entry path (similar to the testing functions in this file 'TestInputManager_Multiselect').
Could you add a function to test this?

**Review by brunasilvazup — CHANGES_REQUESTED** [review]

**Review by lucasdittrichzup — APPROVED** [review]

**Review by brunasilvazup — APPROVED** [review]

**Comment by henriquemoraeszup** [comment]
Hey @joaorufinozup, can you update the description with code example of how to use on the `config.json` for documentation purposes please?

**Comment by brunasilvazup** [comment]
Hi @JoaoDanielRufino,
During the quality control tests in Windows, we identified some bugs with the command `rit add workspace`, we came to the conclusion that it was due to changes in your PR, we will do more tests next week.

That was the reason why your PR did not come out in the last release, if you want to do this test we would also be grateful.

In the meantime I will leave PR out of the way for review and with a wait.

**Review by henriquemoraeszup — CHANGES_REQUESTED** [review]
When you run `--help` the flag of a path input is not mapping. It is also recommended to autofill just enough for the common option, not just pick the first one. Example
```
ls
ritchie-cli    ritchie-henrique
```
if I type `r` then `tab` it should fill up to `ritchie-`

**Comment by joaorufinozup** [comment]
> When you run `--help` the flag of a path input is not mapping. It is also recommended to autofill just enough for the common option, not just pick the first one. Example
> 
> ```
> ls
> ritchie-cli    ritchie-henrique
> ```
> 
> if I type `r` then `tab` it should fill up to `ritchie-`

The `--help`flag is working now

**Review by henriquemoraeszup — APPROVED** [review]

**Pull request merged**